xdsl/dialects/memref.py

```diff
@@ -509,6 +511,40 @@ class ExpandShapeOp(AlterShapeOp):
     name = "memref.expand_shape"
 
 
+@irdl_op_definition
+class ExtractStridedMetaDataOp(IRDLOperation):
+    name = "memref.extract_strided_metadata"
+
+    source: Operand = operand_def(MemRefType)
+
+    base_buffer: OpResult = result_def(MemRefType)
+    offset: OpResult = result_def(IndexType)
+    sizes: VarOpResult = var_result_def(IndexType)
+    strides: VarOpResult = var_result_def(IndexType)
+
+    irdl_options = [AttrSizedResultSegments()]
+
+    @staticmethod
+    def from_source_memref(
+        source: SSAValue | Operation, source_type: MemRefType[Attribute]
+    ):
+        source_shape = source_type.get_shape()
+        # Return a rank zero memref with the memref type
+        base_buffer_type = MemRefType(
+            source_type.element_type,
+            [],
+            NoneAttr(),
+            source_type.memory_space,
+        )
+        offset_type = IndexType()
+        strides_type = [IndexType() for _ in source_shape]
+        sizes_type = [IndexType() for _ in source_shape]
```

Comment:
You could write this as `[IndexType()] * len(source_shape)`, but I don't know if you want to^^